tedana/decomposition/eigendecomp.py

```diff
@@ -175,21 +281,30 @@ def tedpca(catd, OCcatd, combmode, mask, t2s, t2sG, stabilize,
     if wvpca:
         dz, cAl = dwtmat(dz)
 
-    if not op.exists('pcastate.pkl'):
-        voxel_comp_weights, varex, comp_ts = run_svd(dz)
+    fname = op.abspath('pcastate.pkl')
+    if op.exists('pcastate.pkl'):
+        LGR.info('Loading PCA from: pcastate.pkl')
+        with open('pcastate.pkl', 'rb') as handle:
+            pcastate = pickle.load(handle)
+
+        if pcastate['method'] != method:
+            LGR.warning('Method from PCA state file ({0}) does not match '
+                        'requested method ({1}).'.format(pcastate['method'],
+                                                         method))
+            state_found = False
+        else:
+            state_found = True
+    else:
+        state_found = False
+
+    if not state_found:
```

Comment:
You could clean this up as:

```
perform_mle = (method == 'mle')
voxel_comp_weights, varex, comp_ts = run_svd(dz, mle=perform_mle)
```

Reply:
That's a good catch! We need to split run_svd into two functions though, in order to cite the MLE paper iff MLE is used.